### PR TITLE
refactor(cart): remove dependency on driver module

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
`@daffodil/cart` is dependent on `@daffodil/driver`, which makes the driver refactor and removing the `@daffodil/driver` module more difficult.

Fixes: https://github.com/graycoreio/daffodil/issues/152

## What is the new behavior?
`@daffodil/cart` is not dependent on `@daffodil/driver`.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
I removed the shopify driver service because it wasn't correct and it forced the cart module to be dependent on the driver module through the config service.

### CODE REVIEW

File: libs/cart/testing/src/drivers/in-memory/cart-driver.module.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    CommonModule
+  ]
+})
+export class DaffInMemoryCartDriverModule {
+  static forRoot(): ModuleWithProviders {
+    return {
+      ngModule: DaffInMemoryCartDriverModule,
+      providers: [
+        {
+          provide: DaffTestingCartDriverModule,
```
Comment (damienwebdev) [inline libs/cart/testing/src/drivers/in-memory/cart-driver.module.ts]: This should be `DaffCartDriver`

File: libs/cart/testing/src/drivers/testing/cart-driver.module.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    CommonModule
+  ]
+})
+export class DaffTestingCartDriverModule {
+  static forRoot(): ModuleWithProviders {
+    return {
+      ngModule: DaffTestingCartDriverModule,
+      providers: [
+        {
+          provide: DaffInMemoryCartDriverModule,
```
Comment (damienwebdev) [inline libs/cart/testing/src/drivers/testing/cart-driver.module.ts]: This should be `DaffCartDriver`

File: libs/cart/testing/src/drivers/in-memory/cart-driver.module.ts (hunk 1 above)
Comment (lderrickable) [reply libs/cart/testing/src/drivers/in-memory/cart-driver.module.ts]: done

File: libs/cart/testing/src/drivers/testing/cart-driver.module.ts (hunk 2 above)
Comment (lderrickable) [reply libs/cart/testing/src/drivers/testing/cart-driver.module.ts]: done

**Pull request merged**